src/core/TsApi.py

```diff
@@ -4,144 +4,332 @@
 import sys
 import threading
 import time
-from typing import List
+from concurrent.futures import ThreadPoolExecutor, Future
+from queue import PriorityQueue, Empty
+from typing import Dict, List
 
 from pywhispercpp.utils import download_model
 
 from packages.File import File
 from core.Transcriber import Transcriber
 from packages.Default import Default
-from utils.database import Database
+from utils.database import Database, JobStatus
 
 
 class TsApi:
-
+    """Main API class with improved job handling"""
+    
     def __init__(self):
-        """
-        Initialisiert die TsAPI-Klasse und startet den Dienst.
-
-        Initialisiert die Warteschlange und laufende Jobs, lädt das
-        Whisper-Modell und richtet Signalhandler für das Herunterfahren ein.
-        """
+        """Initialize TsAPI with ThreadPoolExecutor and event-driven queue"""
         logging.info("Starting TsAPI...")
-        # Shutdown Handling
+        
+        # Shutdown handling
         signal.signal(signal.SIGTERM, self.exit)
         signal.signal(signal.SIGINT, self.exit)
-        # Creating/Loading Database
+        
+        # Database
         self.database = Database()
-        # Queue and Running Jobs
-        self.running_jobs: List[Default.Entry] = []
-        # Load & Create Module
+        
+        # Thread pool for job execution
+        self.executor = ThreadPoolExecutor(
+            max_workers=int(os.environ.get("parallel_workers", 1)),
+            thread_name_prefix="transcriber"
+        )
+        
+        # Job tracking
+        self.job_futures: Dict[str, Future] = {}
+        self.cancel_events: Dict[str, threading.Event] = {}
+        self.job_lock = threading.Lock()
+        
+        # Priority queue (in-memory, backed by database)
+        self.job_queue = PriorityQueue()
+        
+        # Load queue from database
+        self._load_queue_from_database()
+        
+        # Load & create default file module
         if "DefaultFileModule" not in self.database.modules:
             self.file_module = File(module_uid="DefaultFileModule")
-            self.database.modules["DefaultFileModule"] = self.file_module
+            self.database.add_module(self.file_module)
         else:
             self.file_module = self.database.modules["DefaultFileModule"]
-        # Load Whisper Model
-        model_size = os.environ.get("whisper_model")
-        if not os.path.exists("./data/models/" + model_size + ".pt"):
+        
+        # Load Whisper model
+        model_size = os.environ.get("whisper_model", "tiny")
```

Comment:
Do not forget to change back to the regular mode after testing.
```suggestion
        model_size = os.environ.get("whisper_model")
```